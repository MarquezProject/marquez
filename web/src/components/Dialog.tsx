// SPDX-License-Identifier: Apache-2.0

import * as React from 'react'
import { dialogToggle } from '../store/actionCreators'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'

interface IProps {
  dialogIsOpen: boolean
  dialogToggle: typeof dialogToggle
  ignoreWarning: () => void
  editWarningField?: string
  title?: string
}

export default function AlertDialog(props: IProps) {
  function handleClose() {
    props.dialogToggle('')
  }

  return (
    <div>
      <Dialog open={props.dialogIsOpen}>
        <DialogTitle>{props.title}</DialogTitle>
        <DialogContent>
          <DialogContentText>{props.editWarningField}</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button className='dialogButton' color='secondary' onClick={props.ignoreWarning}>
            Continue
          </Button>
          <Button className='dialogButton' color='primary' onClick={handleClose}>
            Cancel
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}
