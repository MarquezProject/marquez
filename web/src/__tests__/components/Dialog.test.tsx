// SPDX-License-Identifier: Apache-2.0

import * as React from 'react'
import { shallow } from 'enzyme'
import Button from '@material-ui/core/Button'
import Dialog from '../../components/Dialog'

describe('Dialog Component', () => {

  const mockProps = {
    dialogIsOpen: true,
    dialogToggle: dialogToggle,
    ignoreWarning: ignoreWarning
  }

  const ignoreWarning = () => {
    return true
  }

  const dialogToggle = () => {
    return true
  }

  const wrapper = shallow(<Dialog {...mockProps} />)

  it('should render two buttons on the dialog', () => {
    expect(wrapper.find(Button)).toHaveLength(2)
  })

  it('renders a snapshot that matches previous', () => {
    expect(wrapper).toMatchSnapshot()
  })

})
