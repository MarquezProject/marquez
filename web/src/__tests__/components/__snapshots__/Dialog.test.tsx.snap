// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component renders a snapshot that matches previous 1`] = `
<div>
  <WithStyles(ForwardRef(Dialog))
    open={true}
  >
    <WithStyles(ForwardRef(DialogTitle)) />
    <WithStyles(ForwardRef(DialogContent))>
      <WithStyles(ForwardRef(DialogContentText)) />
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        className="dialogButton"
        color="secondary"
      >
        Continue
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        className="dialogButton"
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
